### ISSUE TITLE
findParking - not all results in area are returned

### ISSUE THREAD

**Issue by fl-ma** [AaltoAsia/O-MI#15]
Executing this find parking query:
`<omiEnvelope xmlns="http://www.opengroup.org/xsd/omi/1.0/" version="1.0" ttl="0">
    <call msgformat="odf">
      <msg>
        <Objects xmlns="http://www.opengroup.org/xsd/odf/1.0/">
          <Object>
            <id>ParkingService</id>
            <InfoItem name="FindParking">
              <value type="odf">
                <Objects>
                  <Object type="FindParkingRequest">
                    <id>Parameters</id>
                    <InfoItem type="schema:Distance" name="DistanceFromDestination">
                      <value>3000</value>
                    </InfoItem>
                    <InfoItem type="mv:wantCharging" name="wantCharging">
                            <value>false</value>
                          </InfoItem>
                    <Object required="true" type="schema:GeoCoordinates">
                      <id>Destination</id>
                      <InfoItem name="latitude">
                        <value type="xs:double">48.137180</value>
                      </InfoItem>
                      <InfoItem name="longitude">
                        <value type="xs:double">11.575746</value>
                      </InfoItem>
                    </Object>
                    <Object required="true" type="mv:ElectricVehicle">
                      <id>Vehicle</id> 
                    </Object>
                  </Object>
                </Objects>
              </value>
            </InfoItem>
          </Object>
        </Objects>
      </msg>
    </call>
</omiEnvelope>`

Returns only this facility:
`<Object type="mv:ParkingFacility"><id>Facility_Marienpl_ID</id>`

When increasing the radius to 30000, those facilities are returned:
`<Object type="mv:ParkingFacility"><id>Facility_Disabled_nymp_ID</id>`
`<Object type="mv:ParkingFacility"><id>BMW Research and Technology House</id>`

This skips at least the first facility and there should others be in range as well (see data in attachment).
[munich_all.txt](https://github.com/AaltoAsia/O-MI/files/2324658/munich_all.txt)

**Comment by fl-ma** [comment]
tested with v 1.0.4

**Comment by TK009** [comment]
We could not reproduce this bug. You can try to restart your server to see if it helps.

Is your data available via normal read request? Are there any errors in logs/omi.log?

If you still have this problem, could you provide clear steps to reproduce, starting from fresh copy of O-MI Node.

**Comment by fl-ma** [comment]
Yes, the data is available via normal read.
I've included that output to the issue above. Have you tried inserting that into your O-MI node and issuing the query?

**Comment by L-gan** [comment]
Yes. We started empty node, wrote data in munich_all.txt and used the request. We tested with both 1.0.4 and 1.0.5 and both returned following  parking facilities within 30000 meters from destination: BMW Research and Technology, Facility_Disabled_nymp_ID, Facility_Marienpl_ID, BMW second facility and Facility_Donnersb_ID.

**Comment by fl-ma** [comment]
I resetted our node and added some example data again.
Now I get another error: 
When searching explicitely for charging, I get also parking facilities without chargers returned.
See attachment for details
[20180830 findCharging bug.txt](https://github.com/AaltoAsia/O-MI/files/2335249/20180830.findCharging.bug.txt)
